ably/options_test.go

```diff
@@ -1,58 +1,33 @@
 package ably_test
 
 import (
-	"log"
 	"net/url"
 
 	"github.com/ably/ably-go/ably"
 
 	. "github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/ginkgo"
 	. "github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/gomega"
-	"github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/gomega/gbytes"
 )
 
 var _ = Describe("ClientOptions", func() {
-	var (
-		options *ably.ClientOptions
-		buffer  *gbytes.Buffer
-	)
+	var opts *ably.ClientOptions
 
 	BeforeEach(func() {
-		buffer = gbytes.NewBuffer()
-
-		options = &ably.ClientOptions{
-			Key: "id:secret",
-		}
-
-		options.Prepare()
+		opts = &ably.ClientOptions{}
 	})
 
 	It("parses Key into a set of known parameters", func() {
-		Expect(options.Token).To(Equal("id"))
-		Expect(options.Secret).To(Equal("secret"))
-	})
-
-	Context("when Key is invalid", func() {
-		var old *log.Logger
+		err := opts.SetKey("name:secret")
 
-		BeforeEach(func() {
-			old, ably.Log.Logger = ably.Log.Logger, log.New(buffer, "", log.Lmicroseconds|log.Llongfile)
-			options = &ably.ClientOptions{
-				Key: "invalid",
-			}
-
-			options.Prepare()
-		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(opts.KeyName).To(Equal("name"))
+		Expect(opts.KeySecret).To(Equal("secret"))
+	})
 
-		AfterEach(func() {
-			ably.Log.Logger = old
-		})
+	It("when Key is invalid it returns an error", func() {
```

Comment:
> A context sets some configuration (in this case the fact that we pass an invalid key)

This is not configuration, this is actual test - setting an invalid key. But I'll wrap it into Context for the Ginkgo sake ;)